src/compas_rhino/artists/boxartist.py

```diff
@@ -18,28 +24,152 @@ class BoxArtist(ShapeArtist):
+            artist.draw(show_edges=True, show_faces=False)
+
+        collection = []
+        colors = []
+        for point in pcl.points[len(pcl) // 2:]:
+            box = tpl.copy()
+            box.frame.point = point
+            collection.append(box)
+            colors.append(i_to_rgb(random.random()))
+
+        BoxArtist.draw_collection(collection, colors=colors, layer="Test::BoxArtist", clear=False)
+
     """
 
-    def draw(self):
+    def draw(self, show_vertices=False, show_edges=False, show_faces=True, join_faces=True):
         """Draw the box associated with the artist.
 
+        Parameters
+        ----------
+        show_vertices : bool, optional
+            Default is ``False``.
+        show_edges : bool, optional
+            Default is ``False``.
+        show_faces : bool, optional
+            Default is ``True``.
+        join_faces : bool, optional
+            Default is ``True``.
+
         Returns
         -------
         list
             The GUIDs of the objects created in Rhino.
         """
         vertices = [list(vertex) for vertex in self.shape.vertices]
-        faces = self.shape.faces
-        edges = self.shape.edges
-        points = [{'pos': point, 'color': self.color} for point in vertices]
-        lines = [{'start': vertices[i], 'end': vertices[j], 'color': self.color} for i, j in edges]
-        polygons = [{'points': [vertices[index] for index in face], 'color': self.color} for face in faces]
-        guids = compas_rhino.draw_points(points, layer=self.layer, clear=False, redraw=False)
-        guids += compas_rhino.draw_lines(lines, layer=self.layer, clear=False, redraw=False)
-        guids += compas_rhino.draw_faces(polygons, layer=self.layer, clear=False, redraw=False)
+        guids = []
+        if show_vertices:
+            points = [{'pos': point, 'color': self.color, 'name': self.name} for point in vertices]
+            guids += compas_rhino.draw_points(points, layer=self.layer, clear=False, redraw=False)
+        if show_edges:
+            edges = self.shape.edges
+            lines = [{'start': vertices[i], 'end': vertices[j], 'color': self.color, 'name': self.name} for i, j in edges]
+            guids += compas_rhino.draw_lines(lines, layer=self.layer, clear=False, redraw=False)
+        if show_faces:
+            faces = self.shape.faces
+            if join_faces:
+                guid = compas_rhino.draw_mesh(vertices, faces, layer=self.layer, name=self.name, color=self.color, disjoint=True)
+                guids.append(guid)
+            else:
+                polygons = [{'points': [vertices[index] for index in face], 'color': self.color, 'name': self.name} for face in faces]
+                guids += compas_rhino.draw_faces(polygons, layer=self.layer, clear=False, redraw=False)
         self._guids = guids
         return guids
 
+    @staticmethod
+    def draw_collection(collection,
```

Comment:
the collection methods offer very little advantage over the individual draw methods. since static methods are somewhat frowned upon anyway, perhaps the collection methods are not really necessary...

Reply:
i tend to agree.  it seems that really the only added functionality of `draw_collection` is the grouping, and it seems like that function should be pulled out, perhaps as a static method to the BaseArtist or maybe somewhere else.  it seems reasonable that one may want to group objects of mixed type, ie a box and a mesh or two.